### PR TITLE
Add CodeQL security scans

### PR DESCRIPTION
**Pull request by amanbrar1999**
## Motivation
Follow up to issue https://github.com/open-telemetry/oteps/issues/144

[CodeQL](https://github.com/github/codeql-action) is GitHub's static analysis engine which scans repos for security vulnerabilities. As the project grows and we near GA it might be useful to have a workflow which checks for security vulnerabilities with every PR so we can ensure every incremental change is following best development practices. Also passing basic security checks will also make sure that there aren't any glaring issues for our users.
## Changes
* This PR adds [CodeQL](https://github.com/github/codeql-action) security checks to the repo
* After every run the workflow uploads the results to GitHub. Details on the run and security alerts will show up in the `security` tab of this repo.

**Workflow Triggers**
* daily cron job at 1:30am
* workflow_dispatch (in case maintainers want to trigger a security check manually)

cc- @alolita

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: "Code Scanning - Action"
+
+on:
+  workflow_dispatch:
+  schedule:
+    #        ┌───────────── minute (0 - 59)
```
Comment (anuraaga) [inline .github/workflows/codeql-analysis.yml:6]: Let's remove these they add some noise, and we have a few cron workflows without the comments so lose some consistency.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  CodeQL-Build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Initializes the CodeQL tools for scanning.
```
Comment (anuraaga) [inline .github/workflows/codeql-analysis.yml:25]: Don't need any of these comments they repeat the step names.

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: java
+
+      # Build the code
```
Comment (anuraaga) [inline .github/workflows/codeql-analysis.yml:31]: Incidentally not correct - so yeah don't need these comments :)

**Review by anuraaga — COMMENTED** [review]
Thanks! Just some cleanup nits.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**